adapters/spi/undertow-adapter-spi/src/main/java/org/keycloak/adapters/undertow/UndertowHttpFacade.java

```diff
@@ -196,18 +196,7 @@ public InputStream getInputStream(boolean buffered) {
 
                 inputStream = new BufferedInputStream(exchange.getInputStream());
                 
-                context.setServletRequest(new HttpServletRequestWrapper((HttpServletRequest) servletRequest) {
-                    @Override
-                    public ServletInputStream getInputStream() {
-                        inputStream.mark(0);
-                        return new ServletInputStream() {
-                            @Override
-                            public int read() throws IOException {
-                                return inputStream.read();
-                            }
-                        };
-                    }
-                });
+                context.setServletRequest(UndertowHttpServletRequest.setupServletInputStream(servletRequest, inputStream));
```

Comment:
This might cause some tests to fail because some adapters allow reading the request input twice. For instance, the policy enforcer relies on this when processing claim information points if the application is also reading the input stream.

But we should be able to get failures if that happen.

Reply:
@pedroigor Thanks!
The static method in the class `UndertowHttpServletRequest` is only a wrapper of the old behavior, so IMHO, the logic behind it didn't change. Or did it?

The purpose of this change is to be able to override the class in the testsuite for Undertow Jakarta EE as it requires some other props for the servlet request.